extruct/tool.py

```diff
@@ -1,65 +1,27 @@
 import argparse
 import json
-
-import lxml
 import requests
-from extruct.jsonld import JsonLdExtractor
-from extruct.rdfa import RDFaExtractor
-from extruct.w3cmicrodata import MicrodataExtractor
-from extruct.xmldom import XmlDomHTMLParser
-
+import extruct
 
-def metadata_from_url(url, microdata=True, jsonld=True, rdfa=True):
+def metadata_from_url(url, syntaxes='all'):
     resp = requests.get(url, timeout=30)
     result = {'url': url, 'status': '{} {}'.format(resp.status_code, resp.reason)}
     try:
         resp.raise_for_status()
     except requests.exceptions.HTTPError:
         return result
-
-    parser = XmlDomHTMLParser(encoding=resp.encoding)
-    tree = lxml.html.fromstring(resp.content, parser=parser)
-
-    if microdata:
-        mde = MicrodataExtractor(nested=True)
-        result['microdata'] = mde.extract_items(tree, resp.url)
-
-    if jsonld:
-        jsonlde = JsonLdExtractor()
-        result['json-ld'] = jsonlde.extract_items(tree, resp.url)
-
-    if rdfa:
-        rdfae = RDFaExtractor()
-        result['rdfa'] = rdfae.extract_items(tree, resp.url)
-
+    result.update(extruct.extract(resp.content, url=url, syntaxes=syntaxes))
     return result
 
 
-def main():
+def main(args=None):
     parser = argparse.ArgumentParser(prog='extruct', description=__doc__)
-    parser.add_argument('url', help='The target URL')
-    parser.add_argument(
-        '--microdata',
-        action='store_true',
-        default=False,
-        help='Extract W3C Microdata from the page.',
-    )
-    parser.add_argument(
-        '--jsonld',
-        action='store_true',
-        default=False,
-        help='Extract JSON-LD metadata from the page.',
-    )
-    parser.add_argument(
-        '--rdfa',
-        action='store_true',
-        default=False,
-        help='Extract RDFa metadata from the page.',
-    )
-    args = parser.parse_args()
-
-    if any((args.microdata, args.jsonld, args.rdfa)):
-        metadata = metadata_from_url(args.url, args.microdata, args.jsonld, args.rdfa)
-    else:
-        metadata = metadata_from_url(args.url)
+    arg = parser.add_argument
+    arg('url', help='The target URL')
+    arg('--syntaxes',
+        default='all',
+        help='Either list of microdata syntaxes to use or "all" (syntaxes\
+             available [microdata, microformat, rdfa, opengraph, jsonld])')
```

Comment:
As far as I understand, here ``args.syntaxes`` would always be a string, while other functions look like they expect either an ``"all"`` string or a list of syntax names. Right now it will probably still work with any reasonable string as well, because ``'microdata' in syntaxes`` works correctly for a string, but I think it's better to convert syntaxes to a list (it it's not default "all" value), and provide an example string in valid format in `--syntaxes`` help.

Reply:
And one more thing - since the public API of the tool.py changed, example in the README at the bottom also needs to be updated.